nixos/modules/services/networking/corerad.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.corerad;
+  configFile = pkgs.writeText "corerad.toml" cfg.config;
+in {
+  meta = {
+    maintainers = with maintainers; [ mdlayher ];
+  };
+
+  options.services.corerad = {
+    enable = mkEnableOption "CoreRAD IPv6 NDP RA daemon";
+
+    config = mkOption {
+      example = ''
+        [[interfaces]]
+        name = "eth0"
+        send_advertisements = true
+      '';
+      type = types.lines;
```

Comment:
For NixOS/rfcs#42, having this be `configFile` instead would be better (taking a path instead of a string), because then this later allows to add nice merging behavior without having to deprecate such a `config` option.

Reply:
Thanks for the feedback! I'm pretty new to NixOS and I'm not exactly sure how to apply the changes you're suggesting. Do you have an existing module that uses this pattern that I could reference?

Reply:
It's just changing the type to `types.path`, changing the example to my above comment, removing the `configFile = pkgs.writeText "corerad.toml" cfg.config;` at the top of the file, and using `cfg.configFile` instead in the `ExecStart`.

Another module doing this is e.g. https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/security/shibboleth-sp.nix

Reply:
How do I reconfigure a service defined like this? It seems to dump more complexity onto the end user, by requiring them to create config files outside the module. Or is there a shorthand that I'm missing?

Reply:
@danderson As mentioned above: 

> And with this being a path it's still easy to assign a string with
> 
> ```nix
> configFile = pkgs.writeText "foo" ''
>   [[interfaces]]
>   ...
> ''
> ```

And once a `settings` option is implemented (with https://github.com/NixOS/nixpkgs/pull/75584), this becomes much nicer, e.g. just

```nix
{
  settings.interfaces.name = "eth0";
}
```

Reply:
Ah, sorry, I missed the extra details.

Reply:
Updated.